communicator/ssh/communicator.go

```diff
@@ -467,7 +467,13 @@ func (c *Communicator) scpSession(scpCommand string, f func(io.Writer, *bufio.Re
 		if exitErr, ok := err.(*ssh.ExitError); ok {
 			// Otherwise, we have an ExitErorr, meaning we can just read
 			// the exit status
-			log.Printf("non-zero exit status: %d", exitErr.ExitStatus())
+			msg := fmt.Sprintf("non-zero exit status: %d", exitErr.ExitStatus())
```

Comment:
This is redundant with  `*ssh.ExitError` itself. 
Changing this to simply print the error message should give you almost the same output, with only slightly different text, plus the signal encountered too.